custom/enikshay/two_b_datamigration/management/commands/import_drtb_cases.py

```diff
@@ -448,34 +561,98 @@ def get_episode_case_properties(domain, column_mapping, row):
 
-    cbnaat_lab_name, cbnaat_lab_id = match_location(domain, column_mapping.get_value("cbnaat_lab", row))
-    if cbnaat_lab_name:
-        properties.update({
-            "diagnosing_facility_name": cbnaat_lab_name,
-            "diagnosing_facility_id": cbnaat_lab_id,
-            "diagnosis_test_type_label": "CBNAAT",
-            "diagnosis_test_type_value": "cbnaat",
-        })
-    if get_cbnaat_resistance(column_mapping, row):
-        properties["diagnosis_test_drug_resistance_list"] = "r"
+    properties.update(get_diagnosis_properties(column_mapping, domain, row))
+
+    properties.update(get_reason_for_test_properties(column_mapping, row))
 
-    cbnaat_result_date = column_mapping.get_value("cbnaat_result_date", row)
-    if cbnaat_result_date:
+    ip_to_cp_date = clean_date(column_mapping.get_value("ip_to_cp_date", row))
+    if ip_to_cp_date:
         properties.update({
-            "diagnosis_test_result_date": clean_date(cbnaat_result_date),
-            "date_of_diagnosis": clean_date(cbnaat_result_date),
+            "cp_initiated": "yes",
+            "cp_initiation_date": ip_to_cp_date,
         })
 
     return properties
 
 
+def get_reason_for_test_properties(column_mapping, row):
+    test_reason_map = {
+        # TODO: there is yellow in the mapping sheet. Was sheel waiting on these?
+        # https://docs.google.com/spreadsheets/d/1Pz-cYNvo5BkF-Sta1ol4ZzfBYIQ4kGlZ3FdJgBLe5WE/edit#gid=704385575
+        "1-Failure": (None, "extended_dst", "mdr_rr_failure"),
+        "1. Presumptive TB	diagnosis_dstb": ("presumptive_tb", None, None),
+        "2-Re-treatment case S+at 4th month": (None, "extended_dst", "4mo_culture_positive"),
+        "3-Contact of known MDR TB case": (None, None, None),
+        "4-S+ at diagnosis, re-treatment case": (None, None, None),
+        "4. Presumptive MDRTB, at diagnosis": (None, "mdr_at_diagnosis", None),
+        "5-Any Follow-up S+": (None, None, None),
+        "5. Contact of MDR/RR TB": (None, "contact_of_mdr_rr", None),
+        "6-S-at Diagnosis re-treatment case": (None, None, None),
+        "6. Follow up Sm+ve at end IP": (None, "follow_up_sm_ve_ip", None),
+        "7-HIV TB Case": (None, None, None),
+        "7. Private Referral-Presumptive MDRTB": (None, "private_referral", None),
+        "8. Discordance resolution": (None, "discordance_resolution", None),
+        "10. MDR/RR TB at diagnosis": (None, "extended_dst", "mdr_rr_diagnosis"),
+        "13. Culture reversion": (None, "extended_dst", "culture_reversion"),
+        "14. Failure of MDR/RR TB regimen": (None, "extended_dst", "mdr_rr_failure"),
+        "15. Recurrent case of second line treatment": (None, "extended_dst", "recurrent_second_line_treatment"),
+        "Private": (None, None, None),
+    }
+    value = column_mapping.get_value("reason_for_testing", row)
+    if not value:
+        return {}
+    if value not in test_reason_map:
+        raise Exception("Unexpected test reason: {}".format(value))
```

Comment:
Thanks, yeah I've actually switched to a custom exception in later commits that I haven't pushed yet. I'll open those in a new PR since there have been a lot of changes since I opened this one.